Mathlib/RingTheory/Valuation/Integers.lean

```diff
@@ -133,8 +135,68 @@ theorem le_iff_dvd {x y : O} : v (algebraMap O F x) ≤ v (algebraMap O F y) ↔
   ⟨hv.dvd_of_le, hv.le_of_dvd⟩
 #align valuation.integers.le_iff_dvd Valuation.Integers.le_iff_dvd
 
+theorem isUnit_of_one' {x : O} (hvx : v (algebraMap O F x) = 1) :
+    IsUnit x := by
+  apply isUnit_of_one hv _ hvx
+  apply IsUnit.mk0
+  rw [← v.ne_zero_iff]
+  simp only [hvx, ne_eq, one_ne_zero, not_false_eq_true]
+
 end Integers
 
+instance instLocalRingInteger : LocalRing v.integer where
+  exists_pair_ne := by
+    use 0, 1
+    simp only [ne_eq, zero_ne_one, not_false_eq_true]
+  isUnit_or_isUnit_of_add_one {a} {b} h := by
+    suffices v a = 1 ∨ v b = 1 from
+      match this with
+      | .inl ha => Or.inl <| Valuation.Integers.isUnit_of_one' (integer.integers v) ha
+      | .inr hb => Or.inr <| Valuation.Integers.isUnit_of_one' (integer.integers v) hb
+    by_contra! hab
+    have : v 1 < 1 := by
+      calc
+      _ = v (a + b) := by
+        congr
+        simp only [← Subring.coe_add, h]
+        rfl
+      _ ≤ max (v a) (v b) := by
+        exact map_add v a b
+      _ < 1 := by
+        apply max_lt
+        exact lt_of_le_of_ne a.2 hab.1
+        exact lt_of_le_of_ne b.2 hab.2
+    simp only [_root_.map_one, lt_self_iff_false] at this
```

Comment:
This should be `ValuationRing v.integer`, be provided below `ValuationRing.of_integers`, using that lemma.

Reply:
I'll move it and modify the proof.